commit ef68fdb5179c9ac7efe4a8a18e9a6cf2034e475b
Author: Florian Westphal <fw@strlen.de>
Date:   Fri Dec 28 01:24:43 2018 +0100

    netfilter: nf_conncount: don't skip eviction when age is negative
    
    commit 4cd273bb91b3001f623f516ec726c49754571b1a upstream.
    
    age is signed integer, so result can be negative when the timestamps
    have a large delta.  In this case we want to discard the entry.
    
    Instead of using age >= 2 || age < 0, just make it unsigned.
    
    Fixes: b36e4523d4d56 ("netfilter: nf_conncount: fix garbage collection confirm race")
    Reviewed-by: Shawn Bohrer <sbohrer@cloudflare.com>
    Signed-off-by: Florian Westphal <fw@strlen.de>
    Signed-off-by: Pablo Neira Ayuso <pablo@netfilter.org>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/net/netfilter/nf_conncount.c b/net/netfilter/nf_conncount.c
index 3271a4e00500..8bb4ed85c262 100644
--- a/net/netfilter/nf_conncount.c
+++ b/net/netfilter/nf_conncount.c
@@ -155,7 +155,7 @@ find_or_evict(struct net *net, struct nf_conncount_list *list,
 	const struct nf_conntrack_tuple_hash *found;
 	unsigned long a, b;
 	int cpu = raw_smp_processor_id();
-	__s32 age;
+	u32 age;
 
 	found = nf_conntrack_find_get(net, &conn->zone, &conn->tuple);
 	if (found)
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1006_ef68fdb5179c9ac7efe4a8a18e9a6cf2034e475b.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1006_ef68fdb5179c9ac7efe4a8a18e9a6cf2034e475b.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
